modules/node_modules/@ncigdc/components/AddCaseFilesToCartButton.js

```diff
@@ -0,0 +1,87 @@
+/* @flow */
+
+import React from 'react';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { compose, withState, withProps, withHandlers } from 'recompose';
+
+import Dropdown from '@ncigdc/uikit/Dropdown';
+import ShoppingCartIcon from 'react-icons/lib/fa/shopping-cart';
+import TrashIcon from 'react-icons/lib/fa/trash';
+import { Tooltip } from '@ncigdc/uikit/Tooltip';
+import { addAllFilesInCart, removeFilesFromCart } from '@ncigdc/dux/cart';
+import withFilters from '@ncigdc/utils/withFilters';
+
+const styles = {
+  dropdownContainer: {
+    left: 0,
+    right: 'auto',
+  },
+};
+
+const AddCaseFilesToCartButton = compose(
+  connect(state => ({ cart: state.cart })),
+  withState('isLoading', 'setIsLoading', false),
+  withFilters(),
+  withProps(({ filters }) => ({
+    isFilteringOnFiles: filters && _.some(filters, (item) => {
+      const content = item.hasOwnProperty('content') ? item.content : item;
+      return content.field.indexOf('files.') === 0;
+    }),
+  })),
+  withHandlers({
+    handleDropdownActivate: ({ relay, setIsLoading, filters, isFilteringOnFiles, hasFiles }) => () => {
+      if (!hasFiles) {
+        // this case has no files to fetch.
+        return;
+      }
+      setIsLoading(true);
+      relay.setVariables({
+        isFileDataRequired: true,
+        filesFilters: isFilteringOnFiles ? filters : null,
+      }, (readyState) => _.some([readyState.ready, readyState.aborted, readyState.error]) && setIsLoading(false));
+    },
+  })
+)(({ hasFiles, files = [], filteredFiles = [], dispatch, handleDropdownActivate, isLoading, isFilteringOnFiles }) => (
+  <Dropdown
+    dropdownStyle={styles.dropdownContainer}
+    dropdownClassName={isLoading ? 'hidden' : 'dropdown-menu'}
+    isDisabled={!hasFiles}
+    button={
+      hasFiles
+        ? (
+          <button className="btn btn-default dropdown-toggle fa fa-shopping-cart" style={{ padding: '0 4px' }}>
+            <span className={isLoading ? 'fa fa-spinner fa-spin' : 'caret'} style={{ marginLeft: '0.5rem' }} />
+            <span className="icon-btn-label">Case Actions</span>
+          </button>
+        )
+        : (
+          <Tooltip Component={<span>This case has no files.</span>}>
+            <button className="btn disabled fa fa-shopping-cart" style={{ padding: '0px 17px 0 4px' }} />
+          </Tooltip>
+        )
+    }
+    onActivate={handleDropdownActivate}
+  >
+    {
+      isLoading
+        ? <li> Loading case files... </li>
+        : ([
+          !!files.length && <li onClick={() => dispatch(addAllFilesInCart(files))} aria-label="Add all Case files to the Cart" role="button" key="addAll">
+            <ShoppingCartIcon style={{ marginRight: '1em' }} />Add all Case files to the Cart ({files.length})
+          </li>,
+          !!files.length && <li onClick={() => dispatch(removeFilesFromCart(files))} aria-label="Remove all Case files from the Cart" role="button" key="removeAll">
```

Comment:
I don't think it's a good idea to use an incorrect implementation just because it happens not to break under current assumptions that aren't guaranteed to always be true  

"Remove case files from cart" should "Remove case files from cart", not "remove files from cart, unless files weren't already in the cart, in which case it'll unexpectedly add the files to the cart"